packages/ts/frontend/src/FluxConnection.ts

```diff
@@ -71,6 +71,16 @@ type EndpointInfo = {
   reconnect?(): ActionOnLostSubscription | void;
 };
 
+let atmosphere: Atmosphere.Atmosphere | undefined;
+
+(async () => {
+  if (!import.meta.env?.VITE_SW_CONTEXT) {
+    atmosphere = await import('atmosphere.js').then((module) => module.default);
+  }
+})().catch((e) => {
+  console.error('Failed to load atmosphere.js', e);
+});
```

Comment:
```suggestion
if (!import.meta.env?.VITE_SW_CONTEXT) {
  try {
    atmosphere = await import('atmosphere.js').then((module) => module.default);
  } catch (e: unknown) {
    console.error('Failed to load atmosphere.js', e);
  }
}
```

I'd suggest using top-level await here. Is it possible?

Reply:
Also, we can probably check the presence of the `document` in `globalThis` here, instead of `VITE_SW_CONTEXT`.

Reply:
top-level await is not allowed according to our project configuration.
> Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)

Reply:
Let's avoid top-level await for now.

Reply:
But that creates a possible race condition, and in some cases the `FluxConnection` can possibly be created before the Atmosphere is loaded.